packages/core/src/content/content-client.ts

```diff
@@ -129,4 +136,89 @@ export class ContentClient {
     return response.manyLocale;
   }
+
+  /**
+   * Retrieves all available taxonomies with optional pagination support.
+   * Note: Only taxonomies are paginated. Terms within each taxonomy are returned in full (no term pagination).
+   * @param {object} [options] - Optional pagination options.
+   * @param {number} [options.pageSize] - Optional. Limits the number of taxonomies returned per page. If not provided, defaults to Content Services API default (20).
+   * @param {string} [options.after] - Optional. Cursor for pagination. Used to fetch the next page of taxonomies.
+   * @returns A promise that resolves to a paginated list of taxonomies, including pagination metadata (`hasMore`, `cursor`) for developer-controlled pagination.
+   */
+  async getTaxonomies(options?: {
+    pageSize?: number;
+    after?: string;
+  }): Promise<ContentItemList<Taxonomy>> {
+    debug.content(
+      'Fetching taxonomies (pageSize: %s, after: %s)',
+      options?.pageSize ?? 'API Default',
+      options?.after ?? ''
+    );
+
+    const variables: Record<string, any> = {};
+    if (options?.pageSize !== undefined) variables.pageSize = options.pageSize;
+
+    variables.after = options?.after ?? '';
+
+    const response = await this.get<TaxonomiesQueryResponse>(GET_TAXONOMIES_QUERY, variables);
+    const data = response?.manyTaxonomy;
+
+    return {
+      results: (data?.results ?? []).map((taxonomy) => ({
+        system: taxonomy.system,
+        terms: {
+          results: taxonomy.terms?.results ?? [],
+          cursor: undefined,
+          hasMore: false,
+        },
+      })),
+      cursor: data?.cursor,
+      hasMore: data?.hasMore ?? false,
+    };
+  }
+
+  /**
+   * Retrieves a taxonomy by its ID, with optional pagination support for its terms.
+   * @param {object} options - Options for fetching the taxonomy.
+   * @param {string} options.id - The unique identifier of the taxonomy.
+   * @param {object} [options.terms] - Optional pagination options for terms.
+   * @param {number} [options.terms.pageSize] - Optional. Limits the number of terms returned per page.
+   * @param {string} [options.terms.after] - Optional. Cursor for pagination. Used to fetch the next page of terms.
+   * @returns A promise that resolves to the taxonomy object, including pagination metadata (`hasMore`, `cursor`) for its terms. Returns `null` if the taxonomy is not found.
+   */
+  async getTaxonomy({
+    id,
+    terms,
+  }: {
+    id: string;
+    terms?: {
+      pageSize?: number;
+      after?: string;
+    };
+  }): Promise<Taxonomy | null> {
+    debug.content(
+      'Getting taxonomy for id: %s (termsPageSize: %s, termsAfter: %s)',
+      id,
+      terms?.pageSize ?? 'API Default',
+      terms?.after ?? ''
+    );
+
+    const variables: Record<string, any> = { id };
+    if (terms?.pageSize !== undefined) variables.termsPageSize = terms.pageSize;
+    if (terms?.after !== undefined) variables.termsAfter = terms.after;
```

Comment:
Can it be simplified?
```suggestion
    const variables: Record<string, any> = {
      id,
      termsPageSize: terms?.pageSize,
      termsAfter: terms?.after,
    };
```